### ISSUE TITLE
Linking FrontEnd & BackEnd

### ISSUE THREAD

**Issue by MMDPROJECT** [Navid-Ebadi-2003/spotify_project#48]
@Retriever5500 Man what part exactly are you working on rn?

**Comment by Retriever5500** [comment]
Nothing in particular just examining the code, what are you doing currently?

**Comment by MMDPROJECT** [comment]
> Nothing in particular just examining the code, what are you doing currently?

rn I want to link everything for loading home-page

**Comment by Retriever5500** [comment]
Okay what issues are you facing?

**Comment by MMDPROJECT** [comment]
> > Nothing in particular just examining the code, what are you doing currently?
> 
> rn I want to link everything for loading home-page



> Okay what issues are you facing?

Just wanted to know how to work together in this part of the code?

**Comment by Retriever5500** [comment]
> Just wanted to know how to work together in this part of the code?

Well if you want to do it over issues then one has to try link these two and if an error/problem was faced that couldn't solve it then you post a step to reproduce the bug procedure and a log of the error.

**Comment by MMDPROJECT** [comment]
Alright so here is the first issue we face : XD
```
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such column: passowrd)
	at org.sqlite.core.DB.newSQLException(DB.java:1179)
	at org.sqlite.core.DB.newSQLException(DB.java:1190)
	at org.sqlite.core.DB.throwex(DB.java:1150)
	at org.sqlite.core.NativeDB.prepare_utf8(Native Method)
	at org.sqlite.core.NativeDB.prepare(NativeDB.java:126)
	at org.sqlite.core.DB.prepare(DB.java:264)
	at org.sqlite.core.CorePreparedStatement.<init>(CorePreparedStatement.java:46)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.<init>(JDBC3PreparedStatement.java:32)
	at org.sqlite.jdbc4.JDBC4PreparedStatement.<init>(JDBC4PreparedStatement.java:25)
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:34)
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:226)
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:206)
	at Server.Database.Query.logIn(Query.java:111)
	at Server.Service.executeRequest(Service.java:87)
	at Server.Service.doService(Service.java:60)
	at Server.Service.run(Service.java:36)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

because final String query = """
                SELECT user_id,username,passowrd FROM User
                WHERE username=?;
                """;
it's **passwrd** instead of **password**
@Retriever5500

**Comment by Retriever5500** [comment]
Welp, will fix it. In which branch are you?

**Comment by MMDPROJECT** [comment]
> Welp, will fix it. In which branch are you?

It occured on download/upload @Retriever5500

**Comment by Retriever5500** [comment]
Okay I'm doing something in query and will get to it

**Comment by MMDPROJECT** [comment]
![image](https://github.com/Navid-Ebadi-2003/spotify_project/assets/125287952/6a22dced-8f6d-4eb3-a1bc-2cae0d79bbc7)
Also please look at this picture :
why **target** is not ignored? didn't we merge gitignore change to the download/upload ? @Retriever5500

**Comment by Retriever5500** [comment]
> Okay I'm doing something in query and will get to it

Will make the according changes on download/upload to save time

**Comment by MMDPROJECT** [comment]
> > Okay I'm doing something in query and will get to it
> 
> Will make the according changes on download/upload to save time

Also please Make a pull request from `download/upload` to `dev` and accept other pull requests to update `dev` to the latest changes we do have and to run the program through that branch
@Retriever5500

**Comment by Retriever5500** [comment]
>  why target is not ignored? didn't we merge gitignore change to the download/upload ? @Retriever5500

I fixed this issue in dev

**Comment by MMDPROJECT** [comment]
> > why target is not ignored? didn't we merge gitignore change to the download/upload ? @Retriever5500
> 
> I fixed this issue in dev

Aha so that's because I'm not running the program on branch `dev`

**Comment by Retriever5500** [comment]
Merged dev into download/upload so just pull

**Comment by MMDPROJECT** [comment]
> Merged dev into download/upload so just pull

Isn't that better if we merge all the branches into `dev` and then run the program from there? @Retriever5500

**Comment by Retriever5500** [comment]
Okay, what branch haven't we merged into `dev` yet, other than `download/upload`?

**Comment by MMDPROJECT** [comment]
> Okay, what branch haven't we merged into `dev` yet, other than `download/upload`?

Told [here](https://github.com/Navid-Ebadi-2003/spotify_project/issues/48#issuecomment-1604276775)

**Comment by Retriever5500** [comment]
Oh okay

**Comment by Retriever5500** [comment]
There was no other error?

**Comment by MMDPROJECT** [comment]
> There was no other error?

No, at least for now? 
I will pull and run the program from `dev`

**Comment by MMDPROJECT** [comment]
Man error is still remaining :
```
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such column: passowrd)
	at org.sqlite.core.DB.newSQLException(DB.java:1179)
	at org.sqlite.core.DB.newSQLException(DB.java:1190)
	at org.sqlite.core.DB.throwex(DB.java:1150)
	at org.sqlite.core.NativeDB.prepare_utf8(Native Method)
	at org.sqlite.core.NativeDB.prepare(NativeDB.java:126)
	at org.sqlite.core.DB.prepare(DB.java:264)
	at org.sqlite.core.CorePreparedStatement.<init>(CorePreparedStatement.java:46)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.<init>(JDBC3PreparedStatement.java:32)
	at org.sqlite.jdbc4.JDBC4PreparedStatement.<init>(JDBC4PreparedStatement.java:25)
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:34)
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:226)
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:206)
	at Server.Database.Query.logIn(Query.java:111)
	at Server.Service.executeRequest(Service.java:87)
	at Server.Service.doService(Service.java:60)
	at Server.Service.run(Service.java:36)
	at java.base/java.lang.Thread.run(Thread.java:1623)

```

this is gitBash commands that i used :
```
MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (linking_and_scenes)
$ git pull navid
remote: Enumerating objects: 105, done.
remote: Counting objects: 100% (81/81), done.
remote: Compressing objects: 100% (28/28), done.
remote: Total 47 (delta 19), reused 32 (delta 12), pack-reused 0
Unpacking objects: 100% (47/47), 5.44 KiB | 1024 bytes/s, done.
From github.com:Navid-Ebadi-2003/spotify_project
   c5a68d9..e7a9d70  dev             -> navid/dev
   5c30052..6bec8b5  download/upload -> navid/download/upload
   d36e0e4..5477feb  query           -> navid/query
Already up to date.

MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (linking_and_scenes)
$ git checkout dev
Switched to branch 'dev'
M       src/main/java/DatabaseBackup/Spotify.db
M       target/classes/Client/Controllers/Boxes/MusicThirdBox/music-third-box.css
M       target/classes/Client/Controllers/Boxes/MusicThirdBox/music-third-box.fxml
Your branch is behind 'navid/dev' by 18 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)

MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (dev)
$ git reset --hard HEAD
Unlink of file 'src/main/java/DatabaseBackup/Spotify.db' failed. Should I try again? (y/n) y
Unlink of file 'src/main/java/DatabaseBackup/Spotify.db' failed. Should I try again? (y/n) y
Unlink of file 'src/main/java/DatabaseBackup/Spotify.db' failed. Should I try again? (y/n) y
Updating files: 100% (3/3), done.
HEAD is now at c5a68d9 Update README.md

MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (dev)
$ git log
commit c5a68d95701f9caf02103fb043e224f5652fd430 (HEAD -> dev)
Author: Navid Ebadi <121602329+Navid-Ebadi-2003@users.noreply.github.com>
Date:   Fri Jun 23 15:53:26 2023 +0330

    Update README.md

commit b46a86f7d90249052ff8873d3bdf0a08c0bab06f (template, fx, database, api)
Merge: 673d8e8 25c6e11
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:49:18 2023 +0330

    Merge branch 'dev' of https://github.com/Navid-Ebadi-2003/spotify_project into dev

commit 673d8e837f9b162222dff56ac4c2527f2cd8cf48
Merge: 619ff54 1098e15
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:44:42 2023 +0330

    Merge remote-tracking branch 'origin/template' into dev

commit 619ff54f7ba5136df6cec52d96ef4b2c1b339609
Merge: 2b45310 d36e0e4
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:43:51 2023 +0330

    Merge remote-tracking branch 'origin/query' into dev

commit 2b4531060be0fa325033c380ff9802827d16f3fc
Merge: f3f94d1 5d6f747
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:43:32 2023 +0330

    Resolve conflict

commit f3f94d16db2390bfdc44f02157be01d517563656
Merge: e9819e4 2cfabe7
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:41:15 2023 +0330

    Resolve conflict

commit 25c6e117fce6401e9222e26d4c178a14f1662700
Merge: 5c30052 d36e0e4
Author: Retriever5500 <126422549+Retriever5500@users.noreply.github.com>
Date:   Fri Jun 23 14:30:29 2023 +0330

    Merge pull request #46 from Navid-Ebadi-2003/query

    Add more data in Json

commit d36e0e41140438a28595d34444ab9e340e20d097
Author: Retriever5500 <Retriever5500@gmail.com>
Date:   Fri Jun 23 14:05:13 2023 +0330

    Add creator and artists to get methods

commit 3fe8c8ac33bade0d68c4eeda9ad1f45347f78dc1

MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (dev)
$ git merge navid/dev
Updating c5a68d9..e7a9d70
Fast-forward
 .../java/Client/Controllers/Boxes/BoxBuilder.java  |  77 +++++++++++++++++++
 .../Boxes/MusicMainBox/MusicMainBoxController.java |  16 ++--
 .../Controllers/HomePage/HomePageController.java   |  14 ++++
 .../Controllers/MainPage/MainPageController.java   |  61 ++++++++-------
 src/main/java/Server/Database/Query.java           |  57 +++++++-------
 src/main/java/Server/Service.java                  |  10 ++-
 .../Client/Controllers/AlbumPage/album-page.css    |  62 ---------------
 .../Client/Controllers/AlbumPage/album-page.fxml   |  37 ---------
 .../Client/Controllers/ArtitsPage/artist-page.css  |  62 ---------------
 .../Client/Controllers/ArtitsPage/artist-page.fxml |  61 ---------------
 .../Boxes/MusicThirdBox/music-third-box.css        |  20 -----
 .../Boxes/MusicThirdBox/music-third-box.fxml       |  32 --------
 .../Client/Controllers/LyricsPage/lyrics-page.css  |   9 ---
 .../Client/Controllers/LyricsPage/lyrics-page.fxml |  16 ----
 .../Controllers/PlaylistPage/playlist-page.css     |  62 ---------------
 .../Controllers/PlaylistPage/playlist-page.fxml    |  47 ------------
 .../Client/Controllers/UserPage/user-page.css      |  61 ---------------
 .../Client/Controllers/UserPage/user-page.fxml     |  84 ---------------------
 target/classes/Icons/download-icon.png             | Bin 10086 -> 0 bytes
 target/classes/Icons/liked.png                     | Bin 772 -> 0 bytes
 target/classes/Icons/play-icon.png                 | Bin 3186 -> 0 bytes
 target/classes/Icons/unliked.png                   | Bin 1622 -> 0 bytes
 22 files changed, 177 insertions(+), 611 deletions(-)
 create mode 100644 src/main/java/Client/Controllers/Boxes/BoxBuilder.java
 delete mode 100644 target/classes/Client/Controllers/AlbumPage/album-page.css
 delete mode 100644 target/classes/Client/Controllers/AlbumPage/album-page.fxml
 delete mode 100644 target/classes/Client/Controllers/ArtitsPage/artist-page.css
 delete mode 100644 target/classes/Client/Controllers/ArtitsPage/artist-page.fxml
 delete mode 100644 target/classes/Client/Controllers/Boxes/MusicThirdBox/music-third-box.css
 delete mode 100644 target/classes/Client/Controllers/Boxes/MusicThirdBox/music-third-box.fxml
 delete mode 100644 target/classes/Client/Controllers/LyricsPage/lyrics-page.css
 delete mode 100644 target/classes/Client/Controllers/LyricsPage/lyrics-page.fxml
 delete mode 100644 target/classes/Client/Controllers/PlaylistPage/playlist-page.css
 delete mode 100644 target/classes/Client/Controllers/PlaylistPage/playlist-page.fxml
 delete mode 100644 target/classes/Client/Controllers/UserPage/user-page.css
 delete mode 100644 target/classes/Client/Controllers/UserPage/user-page.fxml
 delete mode 100644 target/classes/Icons/download-icon.png
 delete mode 100644 target/classes/Icons/liked.png
 delete mode 100644 target/classes/Icons/play-icon.png
 delete mode 100644 target/classes/Icons/unliked.png

MMD_PROJECT@DESKTOP-U5QCPTO MINGW64 /d/dev/Java_Projects/Spotify/spotify_project (dev)
$ git reset --hard HEAD
Unlink of file 'src/main/java/DatabaseBackup/Spotify.db' failed. Should I try again? (y/n) y
HEAD is now at e7a9d70 Merge pull request #50 from Navid-Ebadi-2003/linking_and_scenes

```
@Retriever5500

**Comment by Retriever5500** [comment]
Fixed with commit f9c524f8ede018fcbde7a76f2870f79cfd039ca1

**Comment by MMDPROJECT** [comment]
> Fixed with commit [f9c524f](https://github.com/Navid-Ebadi-2003/spotify_project/commit/f9c524f8ede018fcbde7a76f2870f79cfd039ca1)

Wasn't that better if we accepted #51 ? @Retriever5500

**Comment by Retriever5500** [comment]
Oh thought we merged it...
